### PR TITLE
Added Scala API to ob1k's futures and promises

### PR DESCRIPTION
**Pull request by staslev**
Thought it would be nice to have a native Scala API for ob1k.

### CODE REVIEW

File: ob1k-concurrent-scala/pom.xml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      <version>2.10.2</version>
+    </dependency>
+    <dependency>
+      <groupId>org.scalatest</groupId>
+      <artifactId>scalatest_2.10</artifactId>
+      <version>2.1.3</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>com.google.code.findbugs</groupId>
```
Comment (marenzo) [inline ob1k-concurrent-scala/pom.xml]: Why?

File: ob1k-concurrent-scala/pom.xml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      <artifactId>jsr305</artifactId>
+      <version>3.0.0</version>
+    </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
+      <version>1.7.5</version>
```
Comment (marenzo) [inline ob1k-concurrent-scala/pom.xml]: Why not using parents pom version?

File: ob1k-concurrent-scala/pom.xml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-log4j12</artifactId>
+      <version>1.7.5</version>
```
Comment (marenzo) [inline ob1k-concurrent-scala/pom.xml]: same as above

File: ob1k-concurrent-scala/pom.xml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>4.10</version>
```
Comment (marenzo) [inline ob1k-concurrent-scala/pom.xml]: same as above

File: ob1k-concurrent-scala/pom.xml [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+            <goals>
+              <goal>testCompile</goal>
+            </goals>
+          </execution>
+        </executions>
+        <configuration>
+          <recompileMode>incremental</recompileMode>
+          <useZincServer>true</useZincServer>
+          <javacArgs>
+            <!-- use ${java.compiler.source}, ${java.compiler.target} ? -->
```
Comment (marenzo) [inline ob1k-concurrent-scala/pom.xml]: Yes

**Comment by marenzo** [comment]
I'm not sure it should be sit in new module. Maybe together with ob1k-concurrent just under src/main/scala?
@eranharel @haski check out also

File: ob1k-concurrent-scala/pom.xml (hunk 2 above)
Comment (staslev) [inline ob1k-concurrent-scala/pom.xml]: I was asking myself the same question.
Have you taken a look at ob1k-concurrent's pom lately?

File: ob1k-concurrent-scala/pom.xml (hunk 1 above)
Comment (staslev) [inline ob1k-concurrent-scala/pom.xml]: Which one?

Comment (marenzo) [inline ob1k-concurrent-scala/pom.xml]: <groupId>com.google.code.findbugs</groupId>

**Comment by staslev** [comment]
Not sure I see a benefit to mixing Scala related dependencies into the pure Java ob1k-concurrent module. Nevertheless, technically it is an option.

File: ob1k-concurrent-scala/pom.xml (hunk 2 above)
Comment (marenzo) [inline ob1k-concurrent-scala/pom.xml]: Thanks, I'll update the pom later.
But in the meanwhile you're welcome to remove the versions from your pom :)

File: ob1k-concurrent-scala/pom.xml (hunk 1 above)
Comment (staslev) [inline ob1k-concurrent-scala/pom.xml]: https://issues.scala-lang.org/browse/SI-7751

**Comment by marenzo** [comment]
Also true.
We'll talk about it on Sunday

**Pull request merged**